src/input.c

```diff
@@ -38,3 +40,228 @@ char* ReadUserInput(const char* const message) {
+  attributes.train = attribute_train;
+
+  if (attribute_car) {
+    int attribute_car_efficiency = GetInputInteger(
+        "What is the efficiency of the car in (UNIT):", kMaxCarEfficiency);
+    attributes.car = attribute_car;
+    attributes.car_fuel_efficiency = attribute_car_efficiency;
+  } else {
+    attributes.car = false;
+    attributes.car_fuel_efficiency = 0;
+  }
+
+  return attributes;
+}
+
+UserPreferences PresetAttributes(void) {
+  const double kAttributePreferences = 0.25;
+  const bool kPredefinedTransport = true;
+  const int kFuelEfficiency = 100;
+
+  UserPreferences user_attributes_generic = {
+      kAttributePreferences, kAttributePreferences, kAttributePreferences,
+      kAttributePreferences, kPredefinedTransport,  kPredefinedTransport,
+      kPredefinedTransport,  kPredefinedTransport,  kFuelEfficiency};
+
+  return user_attributes_generic;
+}
+
+UserPreferences FileAttributes(void) {
+  const bool kTransportPreferences = true;
+  const int kFuelEfficiency = 100;
+
+  UserPreferences file_attributes = {0};
+
+  file_attributes.price = GetUserPreference("price");
+  file_attributes.time = GetUserPreference("time");
+  file_attributes.environment = GetUserPreference("environment");
+  file_attributes.health = GetUserPreference("health");
+
+  file_attributes.walk = kTransportPreferences;
+  file_attributes.bike = kTransportPreferences;
+  file_attributes.train = kTransportPreferences;
+  file_attributes.car = kTransportPreferences;
+  file_attributes.car_fuel_efficiency = kFuelEfficiency;
+
+  return file_attributes;
+}
+
+UserPreferences PromptAttributes(void) {
+  while (1) {
+    char* choice = ReadUserInput(
+        "Do you want to manually input your preferenes? (m)\nLoad preferences "
+        "from a file (f)\nUse presets? (p)\n       >>>");
+
+    char user_choice = choice[0];
+
+    free(choice);
+
+    switch (user_choice) {
+      case kManually:
+        return ManualAttributes();
+
+      case kPreset:
+        return PresetAttributes();
+
+      case kFile:
+        return FileAttributes();
+
+      default:
+        printf("Invalid input!\n");
+        break;
+    }
+    printf("\n");
+  }
+}
+
+void PromptInputParameters(InputParameters* user_data_struct) {
+  const double kAttributePreferences = 0.25;
+  const bool kTransportPreferences = true;
+  const int kHoursInDay = 24;
+  const int kMinutesInHour = 60;
+
+  const int kFuelEfficiency = 100;
+  const int kClockHour = 12;
+  const int kClockMinutes = 30;
+
+  UserPreferences user_attributes_generic = {
+      kAttributePreferences, kAttributePreferences, kAttributePreferences,
+      kAttributePreferences, kTransportPreferences, kTransportPreferences,
+      kTransportPreferences, kTransportPreferences, kFuelEfficiency};
+
+  char* choose_prompts = ReadUserInput(
+      "Use predefined or manually input? (p = predefined or m = manual) :");
+
+  char choice = choose_prompts[0];
+
+  free(choose_prompts);
+
+  if (choice == 'p') {
```

Comment:
Why not use switch or if/ifelse here? Just curious

Reply:
Mistake from earlier code version. Changed to if / ifelse